### PR TITLE
[IMP] website_quote:add checkbox into quote template to add empty templa...

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
...te in quotation and implement back button into edit quote template
- Task: https://www.odoo.com/web?#id=9298&view_type=form&model=project.task&menu_id=3942&action=327
- Pad: https://pad.odoo.com/p/r.vr8kETxQIzMBoUZA

### CODE REVIEW

File: addons/website_quote/views/website_quotation.xml [hunk 1]
```diff
@@ -507,6 +507,11 @@
                         </div>
                     </div>
                     <div class="col-md-9">
+                        <div class="row">
+                            <div groups="base.group_website_publisher" t-ignore="true" class="css_editable_mode_hidden col-md-offset-11">
```
Comment (tde-banana-odoo) [inline addons/website_quote/views/website_quotation.xml]: What is the link between being a webmaster and being able to come back to the quotation ?

File: addons/website_quote/controllers/main.py [hunk 2]
```diff
@@ -138,7 +138,8 @@ def update(self, line_id, remove=False, unlink=False, order_id=None, token=None,
 
     @http.route(["/quote/template/<model('sale.quote.template'):quote>"], type='http', auth="user", website=True)
     def template_view(self, quote, **post):
-        values = { 'template': quote }
+        dummy, action = request.registry.get('ir.model.data').get_object_reference(request.cr, request.uid, 'website_quote', 'action_sale_quotation_template')
```
Comment (tde-banana-odoo) [inline addons/website_quote/controllers/main.py]: Please use xmlid_to_res_id .

Comment (apa-odoo) [inline addons/website_quote/controllers/main.py]: Done

File: addons/website_quote/views/website_quotation.xml (hunk 1 above)
Comment (apa-odoo) [inline addons/website_quote/views/website_quotation.xml]: @tde-banana-odoo Back button only visible to the login user, but controller have  auth="user". so no require groups, can i remove it?

File: addons/website_quote/models/order.py [hunk 3]
```diff
@@ -242,6 +228,35 @@ def onchange_template_id(self, cr, uid, ids, template_id, partner=False, fiscal_
             'validity_date': date,
             'require_payment': quote_template.require_payment
         }
+        if quote_template.add_lines_in_quote:
+            for line in quote_template.quote_line:
+                res = self.pool.get('sale.order.line').product_id_change(cr, uid, False,
```
Comment (atp-odoo) [inline addons/website_quote/models/order.py]: define self.pool outside of for loop

File: addons/website_quote/models/order.py [hunk 4]
```diff
@@ -26,7 +27,15 @@ class sale_quote_template(osv.osv):
             (0, 'Not mandatory on website quote validation'),
             (1, 'Immediate after website order validation')
             ], 'Payment', help="Require immediate payment by the customer when validating the order from the website quote"),
+        'add_lines_in_quote': fields.selection([
+            (1, 'Display all order lines by default'),
+            (0, 'Select some products among the available products in the order lines')
+            ], 'Sale Order', help="In order to describe all the order lines in the template, add all possible products here below. Those products will be automatically added to each quote linked to this template. To create an empty quote, unselect this box.")
```
Comment (tde-banana-odoo) [inline addons/website_quote/models/order.py]: Isn't it something linked to the description even if not adding lines ?

File: addons/website_quote/models/order.py [hunk 5]
```diff
@@ -233,7 +257,7 @@ def onchange_template_id(self, cr, uid, ids, template_id, partner=False, fiscal_
             }))
         date = False
         if quote_template.number_of_days > 0:
-            date = (datetime.datetime.now() + datetime.timedelta(quote_template.number_of_days)).strftime("%Y-%m-%d")
+            date = (datetime.datetime.now() + datetime.timedelta(quote_template.number_of_days)).strftime(DEFAULT_SERVER_DATE_FORMAT)
```
Comment (tde-banana-odoo) [inline addons/website_quote/models/order.py]: :heart:

File: addons/website_quote/models/order.py [hunk 6]
```diff
@@ -129,6 +138,18 @@ def write(self, cr, uid, ids, values, context=None):
...
         return super(sale_order_line, self).write(cr, uid, ids, values, context=context)
 
+    # The overrided onchange was written in new api, sorry
+    # Take the description on the order template if the product is present in it
+    @api.multi
+    @api.onchange('product_id')
+    def product_id_change(self):
+        domain = super(sale_order_line, self).product_id_change()
+        if self.order_id.template_id:
+            for quote_line in self.order_id.template_id.quote_line:
```
Comment (tde-banana-odoo) [inline addons/website_quote/models/order.py]: self.name = next(quote_line.name for quote_line in self.order_id.template_id.quote_line if quote_line.product_id == self.product_id) could work. Maybe. Just to show you can use next.

File: addons/website_quote/models/order.py [hunk 7]
```diff
@@ -10,6 +10,7 @@
 import openerp.addons.decimal_precision as dp
 from openerp import SUPERUSER_ID
 from openerp.tools.translate import _
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT
```
Comment (tde-banana-odoo) [inline addons/website_quote/models/order.py]: NotImportant: should be before openerp.tools.translate

File: addons/website_quote/models/order.py [hunk 8]
```diff
@@ -129,6 +130,16 @@ def write(self, cr, uid, ids, values, context=None):
         values = self._inject_quote_description(cr, uid, values, context)
         return super(sale_order_line, self).write(cr, uid, ids, values, context=context)
 
+    # The overrided onchange was written in new api, sorry
+    # Take the description on the order template if the product is present in it
+    @api.multi
+    @api.onchange('product_id')
+    def product_id_change(self):
+        domain = super(sale_order_line, self).product_id_change()
+        if self.order_id.template_id.id:
```
Comment (tde-banana-odoo) [inline addons/website_quote/models/order.py]: NotImportant: id is not necessary.

**Pull request merged**